components/gitpod-protocol/go/reconnecting-ws.go

```diff
@@ -214,25 +220,18 @@ func (rc *ReconnectingWebsocket) connect(ctx context.Context) *WebsocketConnecti
 				return nil
 			}
 		}
-		var statusCode int
-		if resp != nil {
-			statusCode = resp.StatusCode
-		}
-		rc.log.WithField("url", rc.url).Info("websocket handshake")
 
+		delay := exp.NextBackOff()
 		rc.log.WithError(err).
 			WithField("url", rc.url).
 			WithField("badHandshakeCount", fmt.Sprintf("%d/%d", rc.badHandshakeCount, rc.badHandshakeMax)).
 			WithField("statusCode", statusCode).
-			Errorf("failed to connect, trying again in %d seconds...", uint32(delay.Seconds()))
+			WithField("delay", delay.String()).
+			Error("failed to connect, trying again...")
```

Comment:
Do we ever stop retrying?

Reply:
if a workspace times out or we are receiving valid bad handshakes, i.e. our client is not compatible somehow when it does make sense to retry.